--- CMakeLists.txt.orig	2013-05-12 23:59:08.351548839 +0200
+++ CMakeLists.txt	2013-05-13 00:01:53.159553252 +0200
@@ -156,6 +156,37 @@
     message("-- Adding -DNDEBUG to definitions")
         add_definitions(-DNDEBUG)
     endif()
+
+elseif (CMAKE_BUILD_TYPE MATCHES Gentoo_Release)
+    message("-- Building in Release mode")
+
+    ## Set the libraries outout directory
+    set( CL_BIN_DIR bin)
+    set( CL_LIB_DIR lib)
+ 
+    set( CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CL_SRC_ROOT}/${CL_LIB_DIR}" )
+    set( CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CL_SRC_ROOT}/${CL_BIN_DIR}" )
+    set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CL_SRC_ROOT}/${CL_LIB_DIR}" )
+    set( CL_LIBPATH "${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}")
+
+    message("-- Executables will be written into ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
+    message("-- Shared Objects will be written into ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
+
+    add_definitions(-DNDEBUG)
+
+elseif (CMAKE_BUILD_TYPE MATCHES Gentoo_Debug)
+    message("-- Building in Debug mode")
+    ## Set the libraries outout directory
+    set( CL_BIN_DIR bin_debug)
+    set( CL_LIB_DIR lib_debug)
+
+    set( CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CL_SRC_ROOT}/${CL_LIB_DIR}" )
+    set( CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CL_SRC_ROOT}/${CL_BIN_DIR}" )
+    set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CL_SRC_ROOT}/${CL_LIB_DIR}" )
+    set( CL_LIBPATH                     "${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}")
+        
+    message("-- Executables will be written into ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
+    message("-- Shared Objects will be written into ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
     
 else ()
     message("-- Building in Release mode")
